### PR TITLE
feat(EG-489): implement consistent sorting on all tables

### PR DESCRIPTION
**Pull request by willholt-tb**
Info on the sorting: https://dept-au.atlassian.net/browse/EG-489?focusedCommentId=18869

### CODE REVIEW

File: packages/front-end/src/app/pages/orgs/edit-user/index.vue [hunk 1]
```diff
@@ -105,6 +110,16 @@
...
+      })
+      .sort((labA, labB) => {
+        // Lab Manager labs first
+        if (labA.LabManager && !labB.LabManager) return -1;
+        if (!labA.LabManager && labB.LabManager) return 1;
+        // then Lab Technician
+        if (labA.LabTechnician && !labB.LabTechnician) return -1;
+        if (!labA.LabTechnician && labB.LabTechnician) return 1;
+        // then sort by name
+        return stringSortCompare(labA.Name, labB.Name);
```
Comment (simondib) [inline packages/front-end/src/app/pages/orgs/edit-user/index.vue:122]: This is great. I have noticed if you grant access, the list doesn't re-sort until a page refresh (although re-sorting works as expected for the roles changing).

Could you perhaps hook in a list refresh to the Grant Access success event?

image

Comment (simondib) [reply packages/front-end/src/app/pages/orgs/edit-user/index.vue:122]: Also this sort function seems the same as used in `packages/front-end/src/app/pages/labs/[labId]/index.vue` - I'd consider abstracting them for re-use to a new sort composable.

File: packages/front-end/src/app/pages/orgs/edit-user/index.vue [hunk 2]
```diff
@@ -85,6 +85,11 @@
    * Filter rows based on search input for laboratory name
    */
   const filteredTableData = computed(() => {
+    if (selectedUserLabsData.value === null) {
+      // wait till selectedUserLabsData is available to prevent jank as the table reorders the rows in front of the user
+      return [];
```
Comment (simondib) [inline packages/front-end/src/app/pages/orgs/edit-user/index.vue:90]: Nice 👍

**Review by simondib — COMMENTED** [review]
Looks great - just some suggestions to consider and maybe add a brief PR description for future reference.

**Comment by willholt-tb** [comment]
> Looks great - just some suggestions to consider and maybe add a brief PR description for future reference.

@simondib yep, good call, I wrote a bit on the ticket but realize I didn't mention it here. Have added to the PR description.

Will also address your comments now 👍

**Comment by willholt-tb** [comment]
@simondib comments addressed, will merge if you think it looks good

**Review by simondib — APPROVED** [review]

**Pull request merged**